# Exploratory Data Analysis

## Purpose

The purpose of this EDA is to explore patterns in the notable people dataset and understand how birth month may relate to broader characteristics such as occupation, geography, gender, and historical time period.

This analysis will focus on identifying interesting trends, checking data quality, exploring missing values, and comparing groups before deciding which variables may be useful for deeper analysis or storytelling.

At this stage, missing values will not be filled automatically. I will first examine where null values appear and decide whether they should be kept, removed, or handled later depending on the column and analysis question.

# Exploratory Analysis of Notable People Dataset

## Project Overview

This notebook explores patterns in a dataset of notable people, focusing on geography, occupation, gender, birth date, awards, and broader groupings of people.

The goal is to identify interesting relationships in the data, compare groups, and look for patterns that may be worth exploring further.

---

## Questions I Want to Explore

### 1. Countries and Occupations

- Do certain countries have more notable people in specific occupation groups?
- Which occupation groups are most common within each country?
- Which countries have the highest diversity of occupation groups?

---

### 2. Gender and Occupation

- What is the overall gender breakdown in the dataset?
- How does gender distribution vary by occupation?
- How does gender distribution vary by broader occupation group?
- Which occupations or occupation groups are most male-dominated, female-dominated, or more balanced?
- Are there occupation groups where gender representation changes over time?

---

### 3. Birth Date, Time Period, and Occupation

- Is there a relationship between birth year and occupation type?
- Are some occupation groups more common among people born in earlier or later historical periods?
- How has the distribution of notable people by occupation changed over time?
- Are certain fields more represented in modern birth cohorts?
- Are there any unusual patterns by birth month or birth season?

---

### 4. Awards and Recognition

- Does award documentation vary by gender?
- Which occupation groups are most likely to have awards recorded?
- Are certain occupations overrepresented among people with awards?

---

### 5. Pattern Discovery

- Are there unexpected combinations of features that appear frequently?
- Can people be grouped into meaningful categories without using a target variable?
- Are there clusters of people who share similar characteristics based on country, occupation group, gender, and birth year?

---

## Ways I Will Analyze the Data

### Descriptive Statistics

I will start by summarizing the dataset with counts, percentages, and basic distributions. This includes looking at the most common countries, occupations, occupation groups, genders, and birth years.

---

### Grouped Comparisons

I will compare categories using grouped summaries. For example, I can compare birth year by occupation group, gender distribution by occupation group, or award documentation by gender.

---

### Cross-Tabulations

I will use cross-tabulations to compare categorical variables, such as country and occupation group, occupation group and gender, or gender and award status. I will look at both raw counts and percentages so large groups do not dominate the results.

---

### Time-Based Analysis

I will create new date features from `date_of_birth`, such as birth year, birth decade, and birth month. This will help explore how occupation and gender patterns change over time.

---

### Visualization

I will use charts to make patterns easier to understand. Possible visualizations include bar charts, stacked bar charts, heatmaps, line charts, histograms, and box plots.

---

### Possible Machine Learning Extension

Most of this notebook is exploratory data analysis. However, I may also try unsupervised learning, such as clustering people based on country, occupation group, gender, and birth year, to see whether meaningful groups appear naturally in the data.

---

## Important Considerations

This dataset represents notable people, not the general population. Patterns should be interpreted as trends within this dataset, not as universal claims about countries, genders, or occupations.

The results may also reflect historical, geographic, gender, documentation, and selection biases.

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [2]:
us_births_1 = pd.read_csv('data/raw/1995-2002.csv')
us_births_2 = pd.read_csv('data/raw/2003-2006.csv')
notable = pd.read_parquet('data/raw/npd_wikidata.parquet')

In [3]:
notable.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 417937 entries, 0 to 417936
Data columns (total 18 columns):
 #   Column                    Non-Null Count   Dtype         
---  ------                    --------------   -----         
 0   wikidata_url              417937 non-null  string        
 1   label                     417937 non-null  string        
 2   name_in_native_languages  100026 non-null  string        
 3   pseudonyms                12879 non-null   string        
 4   sex_or_gender             417526 non-null  string        
 5   date_of_birth             417937 non-null  datetime64[ns]
 6   place_of_birth            416462 non-null  string        
 7   country_of_birth          417937 non-null  string        
 8   date_of_death             115230 non-null  datetime64[ns]
 9   place_of_death            94520 non-null   string        
 10  country_of_death          94570 non-null   string        
 11  citizenships              399770 non-null  string        
 12  oc

In [4]:
notable['occupation_groups'].unique()

<StringArray>
[                                              'Arts & Culture',
                               'Arts & Culture; Public Figures',
                                               'Public Figures',
                           'Public Figures; Science & Academia',
                                           'Science & Academia',
                                                       'Sports',
                                                     'Business',
                                     'Business; Public Figures',
                                     'Arts & Culture; Business',
                                 'Business; Science & Academia',
                           'Arts & Culture; Science & Academia',
                     'Arts & Culture; Business; Public Figures',
           'Arts & Culture; Public Figures; Science & Academia',
                 'Arts & Culture; Business; Science & Academia',
                                   'Science & Academia; Sports',
           

In [5]:
notable['occupations'].unique()

<StringArray>
[                                        'screenwriter; film director; film producer',
                         'actor; stage actor; director; television actor; film actor',
                               'writer; poet; journalist; opinion journalist; critic',
                                                               'politician; diplomat',
                                                              'politician; economist',
                                                                           'musician',
                                        'engineer; mathematician; university teacher',
 'actor; stage actor; film director; television actor; film actor; cabaret performer',
                                        'university teacher; stage actor; film actor',
                                                             'politician; journalist',
 ...
                                                  'actor; aikidoka; judoka; karateka',
                        

In [6]:
notable['occupations'].value_counts(dropna=False).head(20)

occupations
politician                             52208
actor                                  13159
baseball player                        10531
ice hockey player                       9597
basketball player                       9061
singer                                  8173
writer                                  7880
American football player                7555
painter                                 5895
journalist                              5494
lawyer; politician                      5323
military personnel                      4374
musician                                3285
actor; film actor                       3225
businessperson                          2964
actor; television actor; film actor     2598
actor; television actor                 2510
writer; journalist                      2398
composer                                2172
chess player                            2110
Name: count, dtype: Int64

In [7]:
drop_col = notable[['place_of_death','date_of_death', 'country_of_death', 'signature_url', 'image_url',  'date_of_image']]
notable = notable.drop(columns=drop_col)

In [8]:
notable.shape

(417937, 12)

In [9]:
notable.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 417937 entries, 0 to 417936
Data columns (total 12 columns):
 #   Column                    Non-Null Count   Dtype         
---  ------                    --------------   -----         
 0   wikidata_url              417937 non-null  string        
 1   label                     417937 non-null  string        
 2   name_in_native_languages  100026 non-null  string        
 3   pseudonyms                12879 non-null   string        
 4   sex_or_gender             417526 non-null  string        
 5   date_of_birth             417937 non-null  datetime64[ns]
 6   place_of_birth            416462 non-null  string        
 7   country_of_birth          417937 non-null  string        
 8   citizenships              399770 non-null  string        
 9   occupations               417937 non-null  string        
 10  occupation_groups         417937 non-null  string        
 11  awards                    100007 non-null  string        
dtypes:

## Data Quality

Total rows: 417,937 rows and 12 columns. 
Core columns: `label`, `date_of_birth`, `country_of_birth`, `occupations`, and `occupation_groups`, `sex_or_gender`

Missing Values: 
`sex_or_gender` : will drop these since I can't infer - only 411 rows (about 0.1%)

Supplemental features that are not as important for my analysis: `awards`, `citizenships`, `place_of_birth`, `name_in_native_languages`, and `pseudonyms`

In [14]:
notable.isna().mean().sort_values(ascending=False) * 100

pseudonyms                  96.918435
awards                      76.071274
name_in_native_languages    76.066728
citizenships                 4.346827
place_of_birth               0.352924
sex_or_gender                0.098340
wikidata_url                 0.000000
label                        0.000000
date_of_birth                0.000000
country_of_birth             0.000000
occupations                  0.000000
occupation_groups            0.000000
dtype: float64

In [15]:
notable = notable.dropna(subset=['sex_or_gender'])

In [16]:
notable.shape

(417526, 12)

In [18]:
notable['sex_or_gender'].value_counts(dropna=False)

sex_or_gender
male                  311968
female                104673
trans woman              442
non-binary gender        281
trans man                 69
genderfluid               25
travesti                  12
genderqueer               11
agender                    9
intersex                   5
cisgender woman            5
two-spirit                 4
bigender                   4
intersex man               3
transgender                3
intersex woman             2
cisgender man              2
undisclosed gender         2
transmasculine             1
māhū                       1
kathoey                    1
female organism            1
transfeminine              1
faʻafafine                 1
Name: count, dtype: Int64

In [23]:
gender_map = {
    'male': 'male',
    'female': 'female',
    'trans woman': 'trans',
    'trans man': 'trans',
    'transfeminine': 'trans',
    'transmasculine': 'trans',
    'transgender': 'trans',
    'travesti': 'trans',
    'cisgender woman': 'female',
    'cisgender man': 'male',  
    'non-binary gender': 'non-binary / gender diverse',
    'genderfluid': 'non-binary / gender diverse',
    'genderqueer': 'non-binary / gender diverse',
    'agender': 'non-binary / gender diverse',
    'bigender': 'non-binary / gender diverse',
    'two-spirit': 'non-binary / gender diverse',
    'intersex': 'non-binary / gender diverse',
    'intersex man': 'non-binary / gender diverse',
    'intersex woman': 'non-binary / gender diverse',
    'māhū': 'non-binary / gender diverse',
    'kathoey': 'non-binary / gender diverse',
    'fa’afafine': 'non-binary / gender diverse',
    'undisclosed gender': 'undisclosed',
    'female organism': 'female'}

In [25]:
notable['gender_group'] = notable['sex_or_gender'].map(gender_map)

notable['gender_group'].value_counts(dropna=True)

gender_group
male                           311970
female                         104679
trans                             528
non-binary / gender diverse       346
undisclosed                         2
Name: count, dtype: int64